src/iuse_actor.cpp

```diff
@@ -2273,84 +2289,72 @@ iuse_actor *repair_item_actor::clone() const
     return new repair_item_actor( *this );
 }
 
-bool repair_item_actor::handle_components( player &pl, const item &fix,
-    bool print_msg, bool just_check ) const
-{
-    // Entries valid for repaired items
-    std::set<material_id> valid_entries;
-    for( const auto &mat : materials ) {
-        if( fix.made_of( mat ) ) {
-            valid_entries.insert( mat );
-        }
-    }
+bool repair_item_actor::has_components( const item &fix, const inventory &inv, bool alert ) const {
+    std::vector<item_comp> found;
+    return has_components( fix, inv, found, alert );
```

Comment:
This convenience wrapper signals about wrong indirection. Ideally, there shouldn't be unclaimed objects.